src/components/AgenciesPage/components/AgenciesList.tsx

```diff
@@ -1,14 +1,32 @@
-import React from 'react';
+import React, { useEffect } from 'react';
 import { Grid } from '@mui/material';
+import { useDispatch } from 'react-redux';
+import { Dispatch } from '@reduxjs/toolkit';
 import { AgenciesItem } from './components';
-import { agenciesData } from '../mocks/agenciesData';
-
-export const AgenciesList = () => (
-  <Grid container>
-    {agenciesData.map(agency => (
-      <Grid item key={agency.id} xs={12} sm={6} md={4} lg={3}>
-        <AgenciesItem agency={agency} />
-      </Grid>
-    ))}
-  </Grid>
-);
+import { selectAgencies } from '../selectors/selectors';
+import { getAgenciesData } from '../services/getAgenciesData';
+import './AgenciesList.scss';
+
+export const AgenciesList = () => {
+  const dispatch = useDispatch<Dispatch>();
+  const { agencies } = selectAgencies();
+
+  useEffect(() => {
+    getAgenciesData(dispatch);
```

Comment:
I've removed extra func, because I don't need to pass the props
that's why I don't use carrying 

should I rewrite it?  
![image](https://user-images.githubusercontent.com/77783277/168597553-93c238d7-1216-41e5-9054-93fe2b572145.png)

Reply:
yep, in this case, `getAgenciesData` returns Promise that should be resolved

Reply:
yep, in this case, getAgenciesData returns Promise that should be resolved

Reply:
added curring